commit 8c37b7e89247ad7cee60663cc16a1716fafa7df5 (refs/changes/85/2985/2)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-11-22T22:37:07+00:00 (1 year, 11 months ago)
    
    DOCKER-1127 nightly-1/2 docker in maint: "Uncaught AssertionError: ctx.volumeToDeleteUuid (uuid) is required"
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/backends/sdc/volumes.js b/lib/backends/sdc/volumes.js
index 912134c..af0b04f 100644
--- a/lib/backends/sdc/volumes.js
+++ b/lib/backends/sdc/volumes.js
@@ -300,18 +300,19 @@ function deleteVolume(params, options, callback) {
             volapiClient.listVolumes(listVolumesParams, {
                 headers: {'x-request-id': options.reqId}
             }, function volumesListed(volumesListErr, volumes) {
-                var err;
-
-                if (!volumesListErr && volumes) {
-                    if (volumes.length === 0) {
-                        err = new errors.DockerError('Could not find volume '
-                            + 'with name: ' + params.name);
-                    } else if (volumes.length !== 1) {
-                        err = new errors.DockerError('More than one volume '
-                            + 'with name: ' + params.name);
-                    } else {
-                        ctx.volumeToDeleteUuid = volumes[0].uuid;
-                    }
+                var err = volumesListErr;
+
+                if (volumesListErr) {
+                    err = errors.volapiErrorWrap(volumesListErr,
+                        'Could not find volume with name: ' + params.name);
+                } else if (!volumes || volumes.length === 0) {
+                    err = new errors.DockerError('Could not find volume '
+                        + 'with name: ' + params.name);
+                } else if (volumes.length !== 1) {
+                    err = new errors.DockerError('More than one volume '
+                        + 'with name: ' + params.name);
+                } else {
+                    ctx.volumeToDeleteUuid = volumes[0].uuid;
                 }
 
                 next(err);
diff --git a/test/integration/cli-mount-nfs-volume-created-with-docker-create.test.js b/test/integration/cli-mount-nfs-volume-created-with-docker-create.test.js
index d49f764..eb9c75f 100644
--- a/test/integration/cli-mount-nfs-volume-created-with-docker-create.test.js
+++ b/test/integration/cli-mount-nfs-volume-created-with-docker-create.test.js
@@ -17,7 +17,7 @@ var testVolumes = require('../lib/volumes');
 var volumesCli = require('../lib/volumes-cli');
 
 var test = testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-create-duplicate-name.test.js b/test/integration/cli-nfs-shared-volumes-create-duplicate-name.test.js
index 2bffff4..5d27df5 100644
--- a/test/integration/cli-nfs-shared-volumes-create-duplicate-name.test.js
+++ b/test/integration/cli-nfs-shared-volumes-create-duplicate-name.test.js
@@ -18,7 +18,7 @@ var volumesCli = require('../lib/volumes-cli');
 
 var createTestVolume = mod_testVolumes.createTestVolume;
 var test = mod_testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-create-failure.test.js b/test/integration/cli-nfs-shared-volumes-create-failure.test.js
index 25585a8..e3fe90f 100644
--- a/test/integration/cli-nfs-shared-volumes-create-failure.test.js
+++ b/test/integration/cli-nfs-shared-volumes-create-failure.test.js
@@ -33,7 +33,7 @@ var volumesCli = require('../lib/volumes-cli');
 
 var createTestVolume = mod_testVolumes.createTestVolume;
 var test = mod_testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-create-size.test.js b/test/integration/cli-nfs-shared-volumes-create-size.test.js
index 4d83dde..783d3d2 100644
--- a/test/integration/cli-nfs-shared-volumes-create-size.test.js
+++ b/test/integration/cli-nfs-shared-volumes-create-size.test.js
@@ -20,7 +20,7 @@ var volumesCli = require('../lib/volumes-cli');
 
 var createTestVolume = mod_testVolumes.createTestVolume;
 var test = mod_testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-delete-in-use.test.js b/test/integration/cli-nfs-shared-volumes-delete-in-use.test.js
index d01ecbd..b3e9199 100644
--- a/test/integration/cli-nfs-shared-volumes-delete-in-use.test.js
+++ b/test/integration/cli-nfs-shared-volumes-delete-in-use.test.js
@@ -18,7 +18,7 @@ var mod_testVolumesCli = require('../lib/volumes-cli');
 
 var createTestVolume = mod_testVolumesCli.createTestVolume;
 var test = mod_testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-disabled.test.js b/test/integration/cli-nfs-shared-volumes-disabled.test.js
index 7c39ef4..71df3b6 100644
--- a/test/integration/cli-nfs-shared-volumes-disabled.test.js
+++ b/test/integration/cli-nfs-shared-volumes-disabled.test.js
@@ -26,7 +26,6 @@ var errorMeansNFSSharedVolumeSupportDisabled =
     testVolumes.errorMeansNFSSharedVolumeSupportDisabled;
 
 var test = testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
     checkDockerClientSupportsNfsVols: true
 });
 
@@ -138,6 +137,14 @@ test('setup', function (tt) {
                         + app.metadata.experimental_docker_nfs_shared_volumes);
                 }
 
+                if (app.metadata.experimental_docker_nfs_shared_volumes ===
+                    false) {
+                    t.comment('NFS volumes support already disabled, no need '
+                        + 'to turn it off');
+                    t.end();
+                    return;
+                }
+
                 SAPI_CLIENT.put('/applications/' + SAPI_APP, {
                     action: 'update',
                     metadata: {
diff --git a/test/integration/cli-nfs-shared-volumes-docker-880.test.js b/test/integration/cli-nfs-shared-volumes-docker-880.test.js
index 17407f7..e5f660b 100644
--- a/test/integration/cli-nfs-shared-volumes-docker-880.test.js
+++ b/test/integration/cli-nfs-shared-volumes-docker-880.test.js
@@ -37,7 +37,7 @@ var mod_testVolumes = require('../lib/volumes');
 var volumesCli = require('../lib/volumes-cli');
 
 var test = mod_testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-filter-dangling.test.js b/test/integration/cli-nfs-shared-volumes-filter-dangling.test.js
index 8d068fa..c901098 100644
--- a/test/integration/cli-nfs-shared-volumes-filter-dangling.test.js
+++ b/test/integration/cli-nfs-shared-volumes-filter-dangling.test.js
@@ -18,7 +18,7 @@ var mod_testVolumesCli = require('../lib/volumes-cli');
 
 var createTestVolume = mod_testVolumesCli.createTestVolume;
 var test = mod_testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-mount-overrides-local-volume.test.js b/test/integration/cli-nfs-shared-volumes-mount-overrides-local-volume.test.js
index 4983bf6..c2b6ce2 100644
--- a/test/integration/cli-nfs-shared-volumes-mount-overrides-local-volume.test.js
+++ b/test/integration/cli-nfs-shared-volumes-mount-overrides-local-volume.test.js
@@ -17,7 +17,7 @@ var testVolumes = require('../lib/volumes');
 var volumesCli = require('../lib/volumes-cli');
 
 var test = testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes-mounting-mode.test.js b/test/integration/cli-nfs-shared-volumes-mounting-mode.test.js
index bd6942f..3e5f862 100644
--- a/test/integration/cli-nfs-shared-volumes-mounting-mode.test.js
+++ b/test/integration/cli-nfs-shared-volumes-mounting-mode.test.js
@@ -30,7 +30,7 @@ var volumesCli = require('../lib/volumes-cli');
 var createTestVolume = mod_testVolumes.createTestVolume;
 var dockerVersion = common.parseDockerVersion(process.env.DOCKER_CLI_VERSION);
 var test = mod_testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 var VOLAPI_CLIENT = mod_testVolumes.getVolapiClient();
diff --git a/test/integration/cli-nfs-shared-volumes-networks.test.js b/test/integration/cli-nfs-shared-volumes-networks.test.js
index 05c9df7..5a07ae1 100644
--- a/test/integration/cli-nfs-shared-volumes-networks.test.js
+++ b/test/integration/cli-nfs-shared-volumes-networks.test.js
@@ -22,7 +22,7 @@ var ufds = require('ufds');
 var volumesCli = require('../lib/volumes-cli');
 
 var test = testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/integration/cli-nfs-shared-volumes.test.js b/test/integration/cli-nfs-shared-volumes.test.js
index bdad3e8..ab5c05a 100644
--- a/test/integration/cli-nfs-shared-volumes.test.js
+++ b/test/integration/cli-nfs-shared-volumes.test.js
@@ -22,7 +22,7 @@ var common = require('../lib/common');
 var testVolumes = require('../lib/volumes');
 
 var test = testVolumes.createTestFunc({
-    checkTritonSupportsNfs: true,
+    checkTritonSupportsNfsVols: true,
     checkDockerClientSupportsNfsVols: true
 });
 
diff --git a/test/lib/volumes.js b/test/lib/volumes.js
index f4258f2..70f23fc 100644
--- a/test/lib/volumes.js
+++ b/test/lib/volumes.js
@@ -134,7 +134,7 @@ function createTestFunc(options) {
     assert.optionalBool(options.checkDockerClientSupportsNfsVols,
         'options.checkDockerClientSupportsNfsVols');
     assert.optionalBool(options.checkTritonSupportsNfsVols,
-        'options.checkTritonSupportsNfsVol');
+        'options.checkTritonSupportsNfsVols');
 
     var dockerClientSupportsVols =
         process.env.DOCKER_CLI_VERSION !== undefined
